### PR TITLE
 Add CORS headers to downloads enabling WebTorrent (closes #678)

### LINKED ISSUE #678: Add CORS headers to downloads enabling WebTorrent
Howdy! Inspired by the recent downtime, I've been working on a way to make a resilient Node downloader using [WebTorrent](https://github.com/feross/webtorrent). Even if the primary download source becomes temporarily unavailable, the files would continue to be available on the bittorrent network. For my proof-of-concept [Download with WebTorrent Button](https://github.com/wmhilton/download-with-webtorrent-button/) to work with nodejs.org, I think all that is needed is to add Cross-Origin Resource Sharing (CORS) headers to the server config so it is possible to download files in `https://nodejs.org/dist/` via AJAX. I know that sounds weird, but how the torrent would work is if there are no other users seeding a torrent it would request the file from the source (via AJAX) to bootstrap the process. This also enables new torrents to be created automatically for files that don't have torrents the first time the button is clicked.

Obviously this isn't a panacea, because a lot of downloads are from 'nvm', WebTorrent only works on modern browsers that support WebRTC, and there are probably a host of other things to consider. But I love WebTorrent and love resilient distributed networking, and I thought adding CORS headers would be preferable to using a proxy service like [cors-anywhere](https://cors-anywhere.herokuapp.com/). You can see the demo of the button [here](https://wmhilton.com/download-with-webtorrent-button)

With that I'll open the floor to questions!

### PR DESCRIPTION
**Pull request by wmhilton**
See the discussion in #678. Basically, we want to be able to download node using AJAX. 😁

### CODE REVIEW

File: setup/www/resources/config/nodejs.org [hunk 1]
```diff
@@ -55,9 +55,24 @@ server {
...
+            add_header 'Access-Control-Max-Age' 600;
+            add_header 'Content-Type' 'text/plain charset=UTF-8';
+            add_header 'Content-Length' 0;
+            return 204;
         }
+
+        # JSON files are now covered by the general CORS settings directly above
+        #location ~ \.json$ {
+        #    add_header access-control-allow-origin *;
+        #}
```
Comment (feross) [inline setup/www/resources/config/nodejs.org]: Can you delete these commented-out lines?

**Comment by feross** [comment]
cc'ing relevant reviewers: @jbergstroem @maclover7 @rvagg

File: setup/www/resources/config/nodejs.org (hunk 1 above)
Comment (wmhilton) [reply setup/www/resources/config/nodejs.org]: done

**Comment by sam-github** [comment]
Seems like it might be useful, but very stale. Closing unless someone wants to restart this.

**Pull request closed without merging**